src/client/parser.ts

```diff
@@ -0,0 +1,27 @@
+// ----- Imports ----- //
+
+import { Result, Err, Ok } from 'types/result';
+
+
+// ----- Functions ----- //
+
+const parse = (domParser: DOMParser) => (s: string): Result<string, DocumentFragment> => {
```

Comment:
This is only needed to parse the html we get from capi right?

If that's the case, hopefully we won't need this client side... as all the liveblog blocks are text we control?

Reply:
We do control them, but I believe we're currently planning to send them over the wire as a string of HTML (the one that CAPI provides)? If that's the case, we still need to parse them into a data structure in order to pull out what we need for rendering.

Alternatively we could parse them on the server, and send the result to the client as structured data for rendering.

Reply:
Yeah I think doing that server side would be nicer. But this only applies to rehydrated content? so the `bylineHtml` in this example probably won't need rehydrating?

Reply:
The byline won't, no.

For liveblog blocks I guess we may re-hydrate sections of the page to allow for changes to blocks, key events etc.